### ISSUE TITLE
Problem auto-deriving `Beamable` on composed beams.

### ISSUE THREAD

**Issue by vwwv** [tathougies/beam#222]
So it seems the constraints to auto-derive are too tight to allow this:
```
data SuperBeamT subBeamT f = SuperBeamT
       { _field1  :: C f Foo
       , _field2  :: C f Foo
       , _subBeam :: subBeamT f
       } deriving(Generic)

instance (Beamable subBeamT) => Beamable SuperBeamT
```
would yield:
```
Could not deduce (Generic
                          (subBeamT Database.Beam.Schema.Tables.Ignored))
```
should the rules to auto-derive it be relaxed so that example could compile? If so, would such a PR be welcomed?

**Comment by tathougies** [comment]
Such a PR would be welcomed, although I feel it may not be easy to implement the way things are structured.

**Comment by vwwv** [comment]
> Such a PR would be welcomed, although I feel it may not be easy to implement the way things are structured.

Ah, it seems I was over-optimistic then, I thought such a change could be even trivial  (I've just started using the framework and I'm not used to its usage, not to say its internals).

Anyway, I'll try to take a look in case I can help later once I get more used to the code base :)

**Comment by tathougies** [comment]
Fixed by #233.. Thanks for the contribution @vwwv !
